catalyst_voices/packages/catalyst_voices_blocs/lib/src/registration/cubits/wallet_link_cubit.dart

```diff
@@ -4,30 +4,41 @@ import 'package:catalyst_cardano/catalyst_cardano.dart';
 import 'package:catalyst_voices_blocs/catalyst_voices_blocs.dart';
 import 'package:catalyst_voices_models/catalyst_voices_models.dart';
 import 'package:catalyst_voices_shared/catalyst_voices_shared.dart';
-import 'package:flutter/foundation.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
 import 'package:result_type/result_type.dart';
 
-final class WalletLinkController with ChangeNotifier {
+final class WalletLinkCubit extends Cubit<WalletLink> {
+  WalletLinkStage _stage;
   Result<List<CardanoWallet>, Exception>? _wallets;
 
-  void handleEvent(WalletLinkEvent event) {
-    switch (event) {
-      case RefreshCardanoWalletsEvent():
-        unawaited(_refreshCardanoWallets());
+  WalletLinkCubit()
+      : _stage = WalletLinkStage.intro,
+        super(const WalletLink());
+
+  void changeStage(WalletLinkStage newValue) {
+    if (_stage != newValue) {
+      _stage = newValue;
+      emit(_buildState());
     }
   }
 
-  WalletLink buildState(WalletLinkStage stage) {
-    return WalletLink(
-      stage: stage,
-      state: WalletLinkStateData(
-        wallets: _wallets,
-      ),
-    );
+  Future<void> refreshCardanoWallets() async {
+    try {
+      _wallets = null;
+      emit(_buildState());
```

Comment:
Usually blocs/cubit don't keep their own fields but use state for it. State should be a place where wallets or stage are stored. In other projects I would:

```
emit(state.copyWith(wallets: null));

// ...
```

state is a property of cubit, returns the current state

Reply:
Currently wallets are kept in two places, once - privately as field in cubit, secondly as cubit state (WalletLink)